lib/rmt/import_script.rb

```diff
@@ -0,0 +1,161 @@
+require 'csv'
+require 'ostruct'
+
+
+class ImportScript
+  attr_accessor :config
+
+  class ExitScript < StandardError
+  end
+
+  def initialize(config)
+    @config = config
+  end
+
+  def read_csv(file)
+    # set the quote char to something not used to make sure the csv parser is not interfering with the
+    # JSON quoting.
+    CSV.open(File.join(config.data_dir, file + '.csv'), 'r', { col_sep: "\t", quote_char: "\x00" })
+  end
+
+  def import_repositories
+    read_csv('enabled_repos').each do |row|
+      repo_id, = row
+      repo = Repository.find_by(scc_id: repo_id)
+      if repo
+        repo.mirroring_enabled = true
+        repo.save!
+        puts "Enabled mirroring for repository #{repo_id}"
+      else
+        warn "Repository #{repo_id}, perhaps you no longer have a valid subscription for it"
+      end
+    end
+  end
+
+  def import_custom_repositories
+    read_csv('enabled_custom_repos').each do |row|
+      product_id, repo_name, repo_url = row
+
+      repo = Repository.find_by(external_url: repo_url)
+      product = Product.find_by(id: product_id)
+
+      # create the repository if it does not exist yet
+      unless repo
+        local_path = Repository.make_local_path(repo_url)
+        repo = Repository.create!(external_url: repo_url, name: repo_name, local_path: local_path)
+      end
+
+      if product
+        assoc = RepositoriesServicesAssociation.find_by(service: product.service, repository: repo)
+        unless assoc
+          RepositoriesServicesAssociation.create!(service: product.service, repository: repo)
+          puts "Added association between #{repo.name} and product #{product_id}"
+        end
+      else
+        warn "Product #{product_id} not found"
+      end
+    end
+  end
+
+  def import_systems
+    read_csv('systems').each do |row|
+      login, password, hostname = row
+
+      # rubocop:disable Rails/TimeZone
+      System.create!(
+        login: login,
+        password: password,
+        hostname: hostname,
+        registered_at: Time.now
```

Comment:
Why not export the original registration date?

Reply:
What data I should use here? LASTCONTACT from Clients?